abel/tools/circularize.py

```diff
@@ -0,0 +1,236 @@
+# -*- coding: utf-8 -*-
+from __future__ import division
+import numpy as np
+from scipy import ndimage
+from scipy.interpolate import UnivariateSpline
+from scipy.optimize import leastsq
+
+import abel
+
+# see https://github.com/PyAbel/PyAbel/issues/186 for discussion
+
+
+def circularize_image(IM, method="lsq", center=None, radial_range=None,
+                      zoom=1, smooth=0, nslices=32, ref_angle=None, 
+                      inverse=False, return_correction=False):
+    """
+    Remove radial distortion from a velocity-map-image through radial scaling
+    of the Newton-rings to enforce circularity.
+
+    The image is divided into angular slices, adjacent radial intensity
+    profile slices compared to give a radial scaling factor that best
+    aligns common structure.
+
+    The resultant radial scaling factor vs angle is interpolated to apply the
+    correction to the whole image grid. The image is remapped to the
+    correct cartesian grid.
+
+    Parameters
+    ----------
+    IM : numpy 2D array
+
+    method: str
+        method to align slice profiles,
+        "argmax" - compare intensity-profile.argmax() of each radial slice.
+        "lsq" - determine radial scaling factor by minimizing the
+                intensity-prfoile with an adjacent slice intensity-profile.
+
+    center: str, or None
+        pre-center image using PyAbel centering methods,
+        "com", "convolution", "gaussian", "image_center", "slice"
+
+    radial_range: tuple, or None
+        limit slice comparison to the radial range (rmin, rmax).
+
+    zoom: float
```

Comment:
Interesting... Did you have problems with this? Alternatively, you can just pass `reproject_image_to_polar` a smaller value for `dr` in order to improve this. This might be a case where the default `r=1` of `reproject_image_to_polar` is not sufficient.

You made some nice plots about 1 year ago (https://github.com/PyAbel/PyAbel/issues/39#issuecomment-168664008), which show that dr=0.5 looks much better than dr=1.0. Actually, looking at those plots makes it seem like, if we are really trying to line up the profiles, then we may want to use dr=0.1.